lib/resty/statsd.lua

```diff
@@ -0,0 +1,47 @@
+--
+-- Forked from https://github.com/lonelyplanet/openresty-statsd
+--
+local _M = {}
+
+-- this table will be shared per worker process
+-- if lua_code_cache is off, it will be cleared every request
+_M.buffer = {}
```

Comment:
if this is shared how is access to it serialized? I'm assuming openresty does something to control this

Reply:
Yeah, this is a comment from the original source that I will remove. Turning off [`lua_code_cache`](https://github.com/openresty/lua-nginx-module#lua_code_cache) would require you to create an iprepd client on every request or something.

It's a setting that is more for development than anything else.